rust/signed_doc/src/metadata/mod.rs

```diff
@@ -0,0 +1,327 @@
+//! Catalyst Signed Document Metadata.
+use std::fmt::{Display, Formatter};
+
+mod content_encoding;
+mod content_type;
+mod document_id;
+mod document_ref;
+mod document_type;
+mod document_version;
+
+pub use catalyst_types::uuid::{V4 as UuidV4, V7 as UuidV7};
+pub use content_encoding::ContentEncoding;
+pub use content_type::ContentType;
+pub use document_id::DocumentId;
+pub use document_ref::DocumentRef;
+pub use document_type::DocumentType;
+pub use document_version::DocumentVersion;
+
+/// Catalyst Signed Document Content Encoding Key.
+const CONTENT_ENCODING_KEY: &str = "Content-Encoding";
+
+/// Document Metadata.
+///
+/// These values are extracted from the COSE Sign protected header.
+#[derive(Debug, serde::Deserialize)]
+pub struct Metadata {
+    /// Document Type `UUIDv4`.
+    #[serde(rename = "type")]
+    doc_type: DocumentType,
+    /// Document ID `UUIDv7`.
+    id: DocumentId,
+    /// Document Version `UUIDv7`.
+    ver: DocumentVersion,
+    /// Document Payload Content Type.
+    #[serde(default, rename = "content-type")]
+    content_type: ContentType,
+    /// Document Payload Content Encoding.
+    #[serde(default, rename = "content-encoding")]
+    content_encoding: Option<ContentEncoding>,
+    /// Additional Metadata Fields.
+    #[serde(flatten)]
+    extra: Fields,
+    /// Metadata Content Errors
+    #[serde(skip)]
+    content_errors: Vec<String>,
+}
+
+/// Optional Metadata Fields.
+///
+/// These values are extracted from the COSE Sign protected header labels.
+#[derive(Default, Debug, serde::Deserialize)]
+struct Fields {
```

Comment:
I think its worth to move this struct into the separate own file and rename to `AdditionalField` to keep consistent with design spec.